pkg/actions/karpenter/tasks.go

```diff
@@ -0,0 +1,142 @@
+package karpenter
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/kris-nova/logger"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/builder"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/karpenter"
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+)
+
+const (
+	kubernetesTagFormat = "kubernetes.io/cluster/%s"
+)
+
+type karpenterTask struct {
+	info               string
+	stackManager       manager.StackManager
+	cfg                *api.ClusterConfig
+	ec2API             ec2iface.EC2API
+	karpenterInstaller karpenter.InstallKarpenter
+}
+
+func (k *karpenterTask) Describe() string { return k.info }
+func (k *karpenterTask) Do(errs chan error) error {
+	return k.createKarpenterTask(errs)
+}
+
+// NewTasksToInstallKarpenter defines tasks required to create Karpenter
+func NewTasksToInstallKarpenter(cfg *api.ClusterConfig, stackManager manager.StackManager, ec2API ec2iface.EC2API, karpenterInstaller karpenter.InstallKarpenter) *tasks.TaskTree {
+	taskTree := &tasks.TaskTree{Parallel: true}
+	taskTree.Append(&karpenterTask{
+		info:               fmt.Sprintf("create karpenter for stack %q", cfg.Metadata.Name),
+		stackManager:       stackManager,
+		cfg:                cfg,
+		ec2API:             ec2API,
+		karpenterInstaller: karpenterInstaller,
+	})
+	return taskTree
+}
+
+// createKarpenterTask creates Karpenter
+func (k *karpenterTask) createKarpenterTask(errs chan error) error {
+	name := k.makeKarpenterStackName()
+
+	logger.Info("building nodegroup stack %q", name)
+	stack := builder.NewKarpenterResourceSet(k.cfg)
+	if err := stack.AddAllResources(); err != nil {
+		return err
+	}
+	tags := map[string]string{
+		api.KarpenterNameTag: name,
+	}
```

Comment:
might be worth having the karpenter version tag too? I think it will help debugging down the line